### PR TITLE
Fix PAC and add more tests

### PR DESCRIPTION
**Pull request by abbradar**
This fixes a bug in b220b5438f (tested locally) and also adds more tests for PAC (I haven't run it locally yet, so wait for Travis!).

### CODE REVIEW

**Comment by abbradar** [comment]
https://travis-ci.org/The-Compiler/qutebrowser/jobs/188272906#L424 why is this expected?

**Comment by The-Compiler** [comment]
Whoops - seems like I messed this up indeed and didn't notice even while writing the test... 😊 

That test should probably do an `isinstance` check instead indeed.

**Comment by abbradar** [comment]
No problem, I was just unsure if it's genuinely expected somewhere. This is ready now I think (Arch Linux test failure seems unrelated).

**Comment by The-Compiler** [comment]
Sorry for the silence here - I'm busy with exams currently, but I'll take care of stuff in 2 weeks or so. #2257 just reminded me of this :wink:

**Comment by The-Compiler** [comment]
Thanks, and sorry again for messing this up!

File: tests/unit/browser/webkit/network/test_pac.py [hunk 1]
```diff
@@ -125,6 +125,47 @@ def test_invalid_port():
...
 
 
+def test_no_function():
+    with pytest.raises(pac.EvalProxyError):
+        pac.PACResolver("")
+
+
+def test_fail_eval():
+    with pytest.raises(pac.EvalProxyError):
+        pac.PACResolver("{")
```
Comment (The-Compiler) [inline tests/unit/browser/webkit/network/test_pac.py]: FWIW I merged those two into one test with parametrizing.

**Pull request merged**